src/Data/Conduit/Tar/Unix.hs

```diff
@@ -111,6 +111,7 @@ restoreFileInternal lenient fi@FileInfo {..} = do
                     return (either ((excs ++) . pure) (const excs) eExc)
             unless (null excs) $ yield (return (fi, excs))
         FTNormal -> do
+            liftIO $ Dir.createDirectoryIfMissing True $ Posix.takeDirectory fpStr
```

Comment:
Make it create upon `lenient`:
```haskell
when lenient $ liftIO $ Dir.createDirectoryIfMissing True $ Posix.takeDirectory fpStr
```
Same approach probably should be added to `FTSymbolicLink` as well